quantum/backlight/backlight_timer.c

```diff
@@ -0,0 +1,166 @@
+#include "quantum.h"
+#include "backlight.h"
+#include "debug.h"
+
+#ifndef BACKLIGHT_GPT_DRIVER
+#    define BACKLIGHT_GPT_DRIVER GPTD7
```

Comment:
Currently TIM7 is used for audio -- although that may change with the DAC/PWM PR.
Could we perhaps have a check here to see if they're both enabled and both attempting to use the same timer?

Reply:
As with a lot of our defaults, they dont work for every scenario. 7 was already enabled so just picked it....8 is already enabled? or I can swap to 15/16/17?

Reply:
I don't really mind to be honest -- perhaps we leave this as-is and work out a way to handle collisions with the usages by QMK in general.

Maybe we need to move to a pattern like:
`#define BACKLIGHT_GPT_DRIVER_NUM 7`
and
```c
#if BACKLIGHT_GPT_DRIVER_NUM == AUDIO_DRIVER_NUM
#    error Backlight timer and audio driver collide, need to reassign one.
#endif
```

Reply:
Obviously, a longer-term problem, out of scope of this PR.